includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -64,6 +70,56 @@ public function __construct( $args = [] ) {
 		);
 	}
 
+	/**
+	 * Sanitize all embeds on the page to be AMP compatible.
+	 *
+	 * @param Document $dom DOM.
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+		$nodes = $this->get_raw_embed_nodes( $dom );
+
+		if ( null === $nodes ) {
+			// Bail if the embed handler returns null.
+			return;
+		}
+
+		if ( 0 === $nodes->length ) {
+			return;
+		}
+
+		foreach ( $nodes as $node ) {
+			if ( ! $this->is_raw_embed( $node ) ) {
+				continue;
+			}
+			$this->sanitize_raw_embed( $node );
+		}
+	}
+
+	/**
+	 * Determine if the node is indeed a raw embed.
+	 *
+	 * @param DOMElement $node DOM element.
+	 * @return bool True if it is a raw embed, false otherwise.
+	 */
+	protected function is_raw_embed( DOMElement $node ) {
+		return $node->parentNode && $this->amp_tag !== $node->parentNode->nodeName;
```

Comment:
To facilitate using this to loop over a list of arbitrary nodes (e.g. `childNodes`) probably safer to accept a `DOMNode` instead:

```suggestion
	 * @param DOMNode $node DOM node.
	 * @return bool True if it is a raw embed, false otherwise.
	 */
	protected function is_raw_embed( DOMNode $node ) {
		return $node instanceof DOMElement && $node->parentNode && $this->amp_tag !== $node->parentNode->nodeName;
```